platforms/documentation/docs/src/docs/release/notes.md

```diff
@@ -128,6 +129,50 @@ Execution failed for task ':dependencies'.
 > Get more help at https://help.gradle.org.
 ```
 
+#### Better handling of circular references in providers
+
+Before this release, evaluating a provider that contained a cycle in its value assignment would lead to a `StackOverflowError`. Starting with this release, circular references are properly detected and reported.
```

Comment:
```suggestion
Before this release, evaluating a provider that contained a cycle in its value assignment would lead to a `StackOverflowError`. 
Starting with this release, circular references are properly detected and reported.
```